libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -232,6 +232,13 @@ bool SyntaxChecker::visit(FunctionDefinition const& _function)
 				"Use \"constructor(...) { ... }\" instead."
 			);
 	}
+	if (!_function.isImplemented() && !_function.modifiers().empty())
+	{
+		if (v050)
```

Comment:
Is this really a feature and wasn't just a bug?

Reply:
You mean I should make it an error in both cases?

Reply:
I think we may want to do that, unless there was a weird behaviour where contracts implementing such an abstract contract would in turn execute the modifier:
```
contract C {
  function f() owneronly;
}

contract D is C {
  function f() {
    // will this have the owneronly modifier?
  }
}
```

Reply:
Yep, I agree.

Reply:
I think it is the way simpler solution to just disallow modifiers on unimplemented functions.

Reply:
But then we might have contracts with unexpected behaviour that only cause warnings... I think modifiers on unimplemented functions should either be an error immediately or if it is only a warning for now, then it should at least be an error not to use the same modifiers in derived contracts.

Reply:
> if it is only a warning for now, then it should at least be an error not to use the same modifiers in derived contracts.

I think that is a good middleground and I can agree with that.